### PR TITLE
Asciidoctor 2.0.4

### PR DESCRIPTION
**Pull request by hakandilek**
- Asciidoctor 2.0.2
 - Asciidoctor EPUB3 1.5.0.alpha.8
 - Asciidoctor Mathematical 0.3.0
 - Asciidoctor Reveal.js 2.0.0

needs asciidoctor/asciidoctor-mathematical#48 to build

### CODE REVIEW

**Comment by mojavelinux** [comment]
Can you work with @cexbrayat who also opened a PR?

https://github.com/asciidoctor/docker-asciidoctor/pull/88

The Docker image should definitely use 2.0.2 as a minimum.

**Comment by mojavelinux** [comment]
It would be better if the version variable changes was done in a separate PR.

**Comment by hakandilek** [comment]
Oh I just saw the MR from @cexbrayat right after creating the MR.

As I mentioned, it needs asciidoctor/asciidoctor-mathematical#48 and other dependencies to make the tests pass. Which are failing at #88 by the way.

I'm not sure if it's worth the effort to separate the PR for variable changes. Dependency updates are needed anyhow.

**Comment by mojavelinux** [comment]
Thanks.

It's always easier for each PR to focus on one type of change. Otherwise, it just ends up sitting there.

**Comment by cexbrayat** [comment]
@hakandilek I don't mind letting you merge your first PR to this repo 👍 I'll close mine.

**Comment by marcelhuberfoo** [comment]
Would you mind adjusting the base image tag from alpine:3.7 to alpine:3.9?
It would make it easier to install other up-to-date tools one might need.

**Comment by hakandilek** [comment]
> Would you mind adjusting the base image tag from alpine:3.7 to alpine:3.9?

yes, sure.

**Comment by dduportal** [comment]
Nice PR , thanks for respective works @hakandilek  @cexbrayat  !

Open question: given the accumulated amount of dependencies, would it be a good thing to switch to a `Gemfile` to delegate this to a tool aimed at dependency management?

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,22 @@
-FROM alpine:3.7
+FROM alpine:3.9
 
 LABEL MAINTAINERS="Guillaume Scheibel <guillaume.scheibel@gmail.com>, Damien DUPORTAL <damien.duportal@gmail.com>"
 
-ARG asciidoctor_version=1.5.8
+ARG asciidoctor_version=2.0.2
```
Comment (marcelhuberfoo) [inline Dockerfile]:
```suggestion
ARG asciidoctor_version=2.0.3
```

File: Makefile [hunk 2]
```diff
@@ -3,11 +3,23 @@ DOCKER_IMAGE_NAME ?= docker-asciidoctor
 DOCKERHUB_USERNAME ?= asciidoctor
 DOCKER_IMAGE_TEST_TAG ?= $(shell git rev-parse --short HEAD)
 DOCKER_IMAGE_NAME_TO_TEST ?= $(DOCKERHUB_USERNAME)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TEST_TAG)
-ASCIIDOCTOR_VERSION ?= 1.5.8
+ASCIIDOCTOR_VERSION ?= 2.0.2
```
Comment (marcelhuberfoo) [inline Makefile]:
```suggestion
ASCIIDOCTOR_VERSION ?= 2.0.3
```

File: README.adoc [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 This Docker container provides:
 
-* Asciidoctor 1.5.8
+* Asciidoctor 2.0.2
```
Comment (marcelhuberfoo) [inline README.adoc]:
```suggestion
* Asciidoctor 2.0.3
```

File: README.md [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 This Docker container provides:
 
-  - Asciidoctor 1.5.8
+  - Asciidoctor 2.0.2
```
Comment (marcelhuberfoo) [inline README.md]:
```suggestion
  - Asciidoctor 2.0.3
```

**Comment by hakandilek** [comment]
> It seems the version update from 2.0.2 to 2.0.3 is missing in the following files

yup, just pushed. what an active and brilliant community here. love it. 😃

**Comment by marcelhuberfoo** [comment]
There seems to be something wrong when the build wants to get/fetch/install the asciidoctor-mathematical of version 0.3.0?!
Any ideas?
```
ERROR:  Could not find a valid gem 'asciidoctor-mathematical' (= 0.3.0) in any repository
ERROR:  Possible alternatives: asciidoctor-mathematical
```

**Comment by hakandilek** [comment]
> There seems to be something wrong when the build wants to get/fetch/install the asciidoctor-mathematical of version 0.3.0?!

Changes for the new version are merged but not published yet:
asciidoctor/asciidoctor-mathematical#48
https://rubygems.org/gems/asciidoctor-mathematical

It does not have a pipeline to auto publish the changes. Do you have an idea how to get them published to rubgems?

**Comment by marcelhuberfoo** [comment]
> It does not have a pipeline to auto publish the changes. Do you have an idea how to get them published to rubgems?

Sorry, no. I guess this is up to the (initial) publisher. So we need to wait.

**Comment by mojavelinux** [comment]
Let's make sure @ProgramFan is aware that a release has been requested (see https://github.com/asciidoctor/asciidoctor-mathematical/issues/49).

I've also filed an issue to automate the CI builds and releases (see https://github.com/asciidoctor/asciidoctor-mathematical/issues/50).

**Comment by ProgramFan** [comment]
I am happy for the release. But @tstumm is the owner on rubygems.

**Comment by hakandilek** [comment]
I think this is good to go now. 😃 ✔️

**Comment by dduportal** [comment]
Thanks all for your contributions

**Comment by hakandilek** [comment]
> Unable to activate asciidoctor-epub3-1.5.0.alpha.8, because asciidoctor-2.0.4 conflicts with asciidoctor (~> 1.5.0

I suppose we need a runtime test for asciidoctor-epub3 here.

**Comment by cexbrayat** [comment]
@hakandilek That would be great indeed. Do you want to give it a try? Now that we have some tests in https://github.com/asciidoctor/docker-asciidoctor/blob/master/tests/test_suite.bats , it should not be too hard to add hopefully.

**Comment by mojavelinux** [comment]
The new version of asciidoctor-epub3 is now available. See https://github.com/asciidoctor/asciidoctor-epub3/releases/tag/v1.5.0.alpha.9. I verified it works with Asciidoctor 2.

**Comment by mojavelinux** [comment]
And btw, Asciidoctor 2.0.6 is now available ;)

**Comment by dduportal** [comment]
Opened #90 to work on epub3 fix (and 2.0.6 bump): I need help.

Thanks a lot everyone involved in this PR, it's a pleasure to see such vibrant community.

**Pull request merged**